biogrid/libs/biogrid-simulator/src/lib/biogrid/biogrid.spec.ts

```diff
@@ -92,15 +93,49 @@ describe('classes', () => {
     // There are two non full buildings, and the battery has to be refilled however
     // the components are placed on the grid randomly thus we cannot guarantee the battery is refilled
     // Energy may come from the solar panels but two buildings must be refiled
-    expect(Object.keys(action.getSupplyingPaths()).length).toBeGreaterThanOrEqual(2);
+    expect(
+      Object.keys(action.getSupplyingPaths()).length
+    ).toBeGreaterThanOrEqual(2);
```

Comment:
this seems like two separate tests - (1) action = brain.computeAction and expect getSupplyingPaths length, and (2) grid.takeAction

Please split these up :)